ledger/participant-integration-api/src/main/scala/platform/store/backend/UpdateToMeteringDbDto.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend
+
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.participant.state.v2.Update.TransactionAccepted
+import com.daml.ledger.participant.state.{v2 => state}
+import com.daml.lf.data.Time.Timestamp
+
+object UpdateToMeteringDbDto {
+
+  def apply(
+      clock: () => Long = () => Timestamp.now().micros
+  ): Iterable[(Offset, state.Update)] => Vector[DbDto.TransactionMetering] = input => {
+
+    val time = clock()
+
+    if (input.nonEmpty) {
```

Comment:
I get the intention, but it is not strictly needed if line 38 would be ledgerOffset = input.head._1.toHexString, since in the map case we only will be if groupMapReduce is nonempty, which can only be if input is nonempty. I would leave the if clause out for simplicity. upto you

Reply:
Doing it outside the loop makes it more obvious that the value is shared across records.